R/estimate_RIT.R

```diff
@@ -0,0 +1,103 @@
+#' @title estimate RIT score for a student using regression or interpolation
+#' 
+#' @description Given studentid, measurementscale, and a target_date, the 
+#' function will return an estimated score based on selected method 
+#' 
+#' @param mapvizieR_obj mapvizieR object
+#' @param sid target studentid
+#' @param measurescale target subject
+#' @param target_date date of interest, %Y-%m-%d format
+#' @param method which method to use to estimate RIT score
+#' @param foward default is TRUE, set to FALSE if only scores before target_date should be chosen for 'closest' method
+#' 
+#' @export
+
+
+estimate_rit <- function(
+  mapvizieR_obj,
+  sid,
+  measurescale,
+  target_date,
+  method = c('closest','lm','interpolate'),
+  forward = TRUE
+) {
+  
+  # throw an error if method is not given
+  if (missing(method)) {
+    stop('method not given')
+  }
+  
+  # pull out the cdf
+  cdf <- mapvizieR_obj[['cdf']]
+  
+  target_date <- as.Date(target_date)
+  
+  # find the matchings rows
+  if (require(dplyr)) {
+    student <- cdf %>% dplyr::filter(studentid == sid,
+                                     measurementscale == measurescale)
+  } else {
+    student <- subset(cdf, studentid == sid & measurementscale == measurescale)
+  }
+  
+  # return error if student does not have take a test for given measurementscale
+  if (dim(student)[1] == 0) {
+    
+    warning('student does not have a test for given measurementscale')
+    return(NA)
+    
+  } else if (dim(student)[1] == 1) {
+    
+    warning('student only has one test event for given measurementscale')
+    return(student$testritscore[1])
+    
+  }
+  
+  # if target_date is one of the test dates, return that rit score
+  if (target_date %in% student$teststartdate) {
+    return(student$testritscore[student$teststartdate == target_date])
```

Comment:
I think so.  As long as we ensure or filter records in the AssessmentResults (i.e., cdf) table s.t growthmeasuewyn == TRUE, then we should be fine. Otherwise it is possible to have two test events for the same student on the same day in the same subject.



—
Sent from Mailbox

On Sat, Jun 6, 2015 at 4:58 PM, Andrew Martin <notifications@github.com>
wrote:

>> +  # return error if student does not have take a test for given measurementscale
>> +  if (dim(student)[1] == 0) {
>> +    
>> +    warning('student does not have a test for given measurementscale')
>> +    return(NA)
>> +    
>> +  } else if (dim(student)[1] == 1) {
>> +    
>> +    warning('student only has one test event for given measurementscale')
>> +    return(student$testritscore[1])
>> +    
>> +  }
>> +  
>> +  # if target_date is one of the test dates, return that rit score
>> +  if (target_date %in% student$teststartdate) {
>> +    return(student$testritscore[student$teststartdate == target_date])
> @chrishaid we are ensuring that each student only has one test per term and subject, correct?  so there's no situation where a mapvizieR object would ever have more than 1 test for a student on a given date in a given subject?
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/almartin82/mapvizieR/pull/111/files#r31871929